crates/oxc_traverse/src/lib.rs

```diff
@@ -0,0 +1,149 @@
+//! AST traversal with ability to read up the tree from visitors.
+//!
+//! Please see [`traverse_mut`] for an explanation of API.
```

Comment:
These documents explain things really really good! I should read your PRs one by one, they are all treasures.

Reply:
That's incredibly kind of you to say. Thank you.

I do think it's important to write lengthy comments to explain how things work. But I'm a native English speaker, so I have an advantage in that respect - it's quite easy for me to turn thoughts into English words.

If you find anything where the comments don't make sense, or don't fully explain what the code is doing, or why, please do alert me and I'll do my best to improve them.